lib/api/src/types.rs

```diff
@@ -19,56 +19,60 @@ pub type Val = Value<Function>;
 impl StoreObject for Val {
     fn comes_from_same_store(&self, store: &Store) -> bool {
         match self {
-            Self::FuncRef(f) => Store::same(store, f.store()),
+            Self::FuncRef(None) => true,
+            Self::FuncRef(Some(f)) => Store::same(store, f.store()),
             Self::ExternRef(ExternRef::Ref(_)) | Self::ExternRef(ExternRef::Other(_)) => false,
-            Self::ExternRef(ExternRef::Null) => true,
+            Self::ExternRef(ExternRef::Null) => todo!("update this code"),
```

Comment:
I think if it's a null extern ref we can assume it comes from the same store (at least on the way comes from the same store is currently used).

Reply:
Oh yeah, for sure!  I think the reason I did that is because I intend to delete `ExternRef::Null` and pretty much entirely redo how `ExternRef` works! I just haven't started that process yet.

The existing code used `Val::ExternRef(ExternRef::Null)` for null funcrefs too, I needed to update all uses of that because that's now outdated so I made code paths using that panic with `todo!`. This is very much in-development code, I just got went off on 15 tangents while implementing this and haven't come back there yet!

Reply:
Following up on that, I don't think most of this PR is reviewable yet. There are a few things that I think are close enough to how I want them to look that they could be reviewed, but most of it is just figuring things out and I expect substantial changes to most of the code in this PR so far. 

All that is why I delayed making a PR for as long as a I have -- I only made a PR this morning to make it easier for Nick to give me feedback on some of the compiler-llvm code I was writing!